packages/admin/cms-admin/src/generator/future/generateForm/getForwardedGqlArgs.ts

```diff
@@ -0,0 +1,98 @@
+import { IntrospectionField, IntrospectionInputValue, IntrospectionQuery } from "graphql";
+
+import { Prop } from "../generateForm";
+import { FormFieldConfig } from "../generator";
+import { Imports } from "../utils/generateImportsCode";
+
+type GqlArg = { type: string; name: string; queryOrMutationName: string; isInputArgSubfield: boolean };
+
+export function getForwardedGqlArgs({
+    fields,
+    gqlField,
+    gqlIntrospection,
+}: {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    fields: FormFieldConfig<any>[];
+    gqlField: IntrospectionField;
+    gqlIntrospection: IntrospectionQuery;
+}): {
+    imports: Imports;
+    props: Prop[];
+    gqlArgs: GqlArg[];
+} {
+    const imports: Imports = [];
+    const props: Prop[] = [];
+    const gqlArgs: GqlArg[] = [];
+
+    const skipGqlInputArgFields = fields.map((field) => String(field.name));
+
+    getArgsIncludingInputArgSubfields(gqlField, gqlIntrospection).forEach((arg) => {
+        if (arg.isInputArgSubfield && skipGqlInputArgFields.includes(arg.name)) return;
+
+        if (arg.type === "ID" || arg.type === "String" || arg.type === "DateTime") {
+            props.push({ name: arg.name, optional: false, type: "string" });
+        } else if (arg.type === "Boolean") {
+            props.push({ name: arg.name, optional: false, type: "boolean" });
+        } else if (arg.type === "Int" || arg.type === "Float") {
+            props.push({ name: arg.name, optional: false, type: "number" });
+        } else if (arg.type === "JSONObject") {
+            props.push({ name: arg.name, optional: false, type: "unknown" });
+        } else {
+            props.push({ name: arg.name, optional: false, type: `GQL${arg.type}` }); // generated types contain GQL prefix
+            imports.push({ name: `GQL${arg.type}`, importPath: "@src/graphql.generated" });
+        }
+        gqlArgs.push({ name: arg.name, type: arg.type, queryOrMutationName: arg.gqlField.name, isInputArgSubfield: arg.isInputArgSubfield });
```

Comment:
Why do I need `queryOrMutationName`? Shouldn't the code be applied to a single GraphQL operation anyway?

Reply:
because initially I checked the create and update mutations so I needed to know the origin of this argument to apply it to the correct mutations. additionally generate grid uses practically the same gqlArgs idea with the same type-property-naming therefore the naming **query**OrMutationName...

In a later refactoring I could rename this field to operationName, because I learned query and mutations are both operations...

I didn't came up with an idea where an update-mutation could use a required gql arg so I decided to don't check the update-mutation for now.

Do you want me to remove this field?

Reply:
What do you think about passing `gqlOperation` as argument for `getForwardedGqlArgs` instead? Then the function could be called twice for create and update mutations (if ever needed), and we could remove the filter of the args based on the operation name. IMO this would be easier to understand.

Reply:
good idea, required special duplicate-handling either way needs to be done where the code is rendered like in `generateFormPropsCode(...)`.

Reply:
> What do you think about passing gqlOperation as argument

`gqlField` is already the gqlOperation, but the type of an gqlOperation is `IntrospectionField` so I named it `gqlField`. But I think `gqlOperation` as name fits better.